pydantic_ai_slim/pydantic_ai/models/gemini.py

```diff
@@ -251,30 +267,22 @@ async def _make_request(
             yield r
 
     def _process_response(self, response: _GeminiResponse) -> ModelResponse:
-        vendor_details: dict[str, Any] | None = None
-
         if len(response['candidates']) != 1:
             raise UnexpectedModelBehavior('Expected exactly one candidate in Gemini response')  # pragma: no cover
+        finish_reason_key = response['candidates'][0].get('finish_reason', None)
+        finish_reason = _FINISH_REASONS.get(finish_reason_key, finish_reason_key)
         if 'content' not in response['candidates'][0]:
-            if response['candidates'][0].get('finish_reason') == 'SAFETY':
+            if finish_reason_key == 'SAFETY':
                 raise UnexpectedModelBehavior('Safety settings triggered', str(response))
             else:
                 raise UnexpectedModelBehavior(  # pragma: no cover
                     'Content field missing from Gemini response', str(response)
                 )
         parts = response['candidates'][0]['content']['parts']
-        vendor_id = response.get('vendor_id', None)
-        finish_reason = response['candidates'][0].get('finish_reason')
-        if finish_reason:
-            vendor_details = {'finish_reason': finish_reason}
         usage = _metadata_as_usage(response)
         usage.requests = 1
         return _process_response_from_parts(
-            parts,
-            response.get('model_version', self._model_name),
-            usage,
-            vendor_id=vendor_id,
-            vendor_details=vendor_details,
+            parts, response.get('model_version', self._model_name), usage, finish_reason
```

Comment:
It looks like we accidentally dropped the `vendor_id`!

Reply:
Also for backward compatibility reasons, let's keep the untouched `finish_reason` on `vendor_detail` as well.